### PR TITLE
Add deno

### PR DESCRIPTION
**Pull request by heavenshell**
Add `deno`.

https://user-images.githubusercontent.com/56591/107849075-0b344480-6e3c-11eb-8495-5294b49945f1.mp4

### CODE REVIEW

File: settings/deno.vim [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      \   'enable': v:true,
+      \   'lint': v:true,
+      \   'unstable': v:true,
+      \   'codeLens': {
+      \     'references': v:true,
+      \     'referencesAllFunctions': v:true,
+      \   },
+      \ }),
+      \ 'allowlist': lsp_settings#get('deno', 'allowlist', ['typescript', 'javascript', 'typescriptreact', 'javascriptreact']),
+      \ 'blocklist': lsp_settings#get('deno', 'blocklist', {c->empty(lsp_settings#root_path(['node_modules'])) ? [] : ['typescript', 'javascript', 'typescriptreact', 'javascriptreact']}),
```
Comment (heavenshell) [inline settings/deno.vim:17]: If project directory contains `node_modules`, the project is node project and should be use typescript-language-server, etc.

File: installer/install-deno.cmd [hunk 2]
```diff
@@ -0,0 +1,6 @@
+@echo off
+
+setlocal
+curl -L -o deno-x86_64-pc-windows-msvc.zip "https://github.com/hashicorp/terraform-ls/releases/latest/download/deno-x86_64-pc-windows-msvc.zip"
```
Comment (heavenshell) [inline installer/install-deno.cmd:4]: always download `latest` version.

File: installer/install-deno.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+set -e
+
+os=$(uname -s | tr "[:upper:]" "[:lower:]")
+
+case $os in
+linux)
+  filename="deno-x86_64-unknown-linux-gnu.zip"
+  ;;
+darwin)
+  if [ $(uname -m) == "x86_64" ]; then
```
Comment (heavenshell) [inline installer/install-deno.sh:12]: Determin `intel mac` or `m1 mac`.

Comment (mattn) [reply installer/install-deno.sh:12]: Thank you. Is this a comment for your todo? or explaining for me?

Comment (heavenshell) [reply installer/install-deno.sh:12]: oh, sorry for confusion, just explaining for reviewer 😉

File: installer/install-deno.cmd (hunk 2 above)
Comment (yaegassy) [reply installer/install-deno.cmd:4]: `hashicorp/terraform-ls`? You may need to make changes to `denoland/deno`.

Comment (heavenshell) [reply installer/install-deno.cmd]: oops 😨 
Thanks, fixed 🙇

**Comment by mattn** [comment]
Thank you.

**Pull request merged**